### PR TITLE
Fix `NaiveSyncBatchNorm1d` and `NaiveSyncBatchNorm2d`

### PR DESCRIPTION
**Pull request by jshilong**
## Motivation
The `NaiveSyncBatchNorm1d` and `NaiveSyncBatchNorm2d` in the mmdet3d has two potential bugs.
- Could not do the training and inference when dist is unavailable
- `NaiveSyncBatchNorm1d` does not support the input with shape (N, C), which is not consistent with `nn.BatchNorm1d`

## Modification
- Add `using_dist` flag 
- Add flag `is_two_dim` to support the shape  (N, C),

## BC-breaking (Optional)
None

### CODE REVIEW

**Review by Tai-Wang — APPROVED** [review]

File: mmdet3d/ops/norm.py [hunk 1]
```diff
@@ -53,11 +53,27 @@ def __init__(self, *args, **kwargs):
...
             f'input should be in float32 type, got {input.dtype}'
-        if dist.get_world_size() == 1 or not self.training:
+        using_dist = dist.is_available() and dist.is_initialized()
+        if (not using_dist) or dist.get_world_size() == 1 \
+                or not self.training:
             return super().forward(input)
         assert input.shape[0] > 0, 'SyncBN does not support empty inputs'
+        is_two_dim = input.dim() == 2
+        if is_two_dim:
+            input.unsqueeze_(2)
```
Comment (ZwwWayne) [inline mmdet3d/ops/norm.py:75]: if we want to use inplace operation here, we should also use it in line 97.

Comment (ZwwWayne) [inline mmdet3d/ops/norm.py:75]: if we want to use inplace operation here, we should also use it in line 97.
